airflow/www/static/js/dag/details/taskDuration/index.tsx

```diff
@@ -0,0 +1,143 @@
+import ReactECharts, { ReactEChartsProps } from "src/components/ReactECharts";
+
+const TaskDuration = () => {
+  const {
+    selected: { taskId },
+    onSelect,
+  } = useSelection();
+
+  const {
+    data: { groups },
+  } = useGridData();
+  var maxDuration = 0;
+  var unit = "seconds";
+  var factor = 1;
+
+  const taskInstances = groups.children.filter(
+    (instance) => instance.id === taskId
+  )[0].instances;
+  taskInstances.forEach((instance) => {
+    const runDuration =
+      instance?.startDate && instance?.endDate
+        ? getDuration(instance?.startDate, instance?.endDate) / 1000
+        : 0;
+    const queuedDuration =
+      instance?.queuedDttm && instance?.startDate > instance.queuedDttm
+        ? getDuration(instance?.queuedDttm, instance?.startDate) / 1000
+        : 0;
+
+    if (runDuration > maxDuration) {
+      maxDuration = runDuration;
+    }
+
+    instance.runDuration = runDuration;
+    instance.queuedDuration = queuedDuration;
+  });
+
+  if (maxDuration <= 60 * 2) {
+    factor = 1;
+    unit = "seconds";
+  } else if (maxDuration <= 60 * 60 * 2) {
+    factor = 60;
+    unit = "minutes";
+  } else if (maxDuration <= 24 * 60 * 60 * 2) {
+    factor = 60 * 60 * 24;
+    unit = "days";
+  }
+
+  taskInstances.forEach((instance) => {
+    instance.runDuration = instance.runDuration / factor;
+    instance.queuedDuration = instance.queuedDuration / factor;
+  });
+
+  function tooltipCallback(args) {
+    const { data } = args[0];
+    const { runId } = data;
+    const { taskId } = data;
+    onSelect({ taskId, runId });
+
+    return `
+Run ID : ${data.runId} <br>
+Queued : ${formatDateTime(data.queuedDttm)} <br>
+Started : ${formatDateTime(data.startDate)} <br>
+Ended : ${formatDateTime(data.endDate)} <br>
+Queued Duration : ${data.queuedDuration.toFixed(2)} ${unit}<br>
+Run Duration : ${data.runDuration.toFixed(2)} ${unit}<br>
+Total : ${(data.queuedDuration + data.runDuration).toFixed(2)} ${unit}<br>
+`;
+  }
+
+  const option: ReactEChartsProps["option"] = {
+    series: [
+      {
+        type: "bar",
+        barMinHeight: 1,
+        itemStyle: {
+          color: (params) => stateColors["queued"],
+        },
+        stack: "x",
+      },
+      {
+        type: "bar",
+        barMinHeight: 1,
+        itemStyle: {
+          color: (params) => stateColors[params.data.state],
+        },
+        stack: "x",
+      },
+    ],
+    dataset: {
+      dimensions: ["runId", "queuedDuration", "runDuration"],
+      source: taskInstances,
+    },
+    tooltip: {
+      trigger: "axis",
+      formatter: tooltipCallback,
+      axisPointer: {
+        type: "shadow",
+      },
+    },
+    xAxis: {
+      type: "category",
+      show: false,
+    },
+    yAxis: {
+      type: "value",
+      name: `duration (${unit})`,
+    },
+  };
+
+  const events = {
+    click: (params) => console.log(params),
+  };
+
+  var chart = <ReactECharts option={option} events={events} />;
```

Comment:
Let's just render the chart inside of `<Box>`. I don't see a need for this to be declared as a variable first.

Reply:
I had to use Box to have height of 50% . Making the chart for full page was not looking good IMO especially when there are like 5 runs. We kept it to 50% height but would be happy to remove the box. Removed the variable. Thanks.